컴포넌트 초기화 및 SASS 관련 설정

voidmatcha opened the pull request:
- 회의 때 정의한 컴포넌트 추가:
  - atoms: Logo, Title, Text, TextArea, Button, Emoji, Image
  - components: Header, QuestionBox, Body1, 2(.gitkeep으로 처리 및 추후 네이밍 변경)
- PostCSS 설정
- 전역 SASS 설정
- Vite에서 SASS import 설정
 
**references**
- PostCSS -> Vite: https://vitejs.dev/guide/features.html#postcss
- SASS import -> Vite: https://github.com/vitejs/vite/issues/832

voidmatcha committed:
feat(component): 회의 때 정의한 컴포넌트 생성 및 초기화

- 컴포넌트명으로 Text 사용하는 것에 대해 lint rule 추가(@typescript-eslint/no-redeclare)
- atom과 component로 구분
- 애매한 컴포넌트명은 .gitkeep으로 폴더만 유지

voidmatcha committed:
feat: vite에서 PostCSS 사용할 수 있도록 설정 추가

- postcss.config.js 추가 및 관련 모듈 설치

voidmatcha committed:
feat: 전역 SASS 설정 및 vite 전역 import 설정

- reference: https://github.com/vitejs/vite/issues/832

SeonHyungJo commented:
컴포넌트 폴더에 index.ts와 {컴포넌트명}.tsx 가 2개인 이유가 있을까요? @dididy

SeonHyungJo commented:
전역 SCSS가 별도의 폴더에 담기는게 좋아보입니다. @dididy

voidmatcha commented:
@SeonHyungJo index.tsx에 컴포넌트 코드가 들어가도록 처리하였습니다. style을 담을 별도의 폴더는 styles로 만들었습니다.

voidmatcha committed:
refactor: 사용하지 않아도 될 컴포넌트 내부 index.ts 제거

- 선형님께서 요청하신 사항

voidmatcha committed:
refactor: scss 파일들 styles 폴더 내부로 이동 및 module alias 설정 추가

- 선형님이 요청하신 부분
- styels 경로 module alias는 @styles로 사용 가능

voidmatcha committed:
refactor: refactor: 회의에서 짚어주신 컴포넌트 구성형태 올바르게 수정

- index.tsx에 컴포넌트 코드를 넣도록 함
- 컴포넌트 모듈을 가져와서 <C.Components /> 형태로 사용하는건 Tree shaking시 문제가 될 수 있음
  따라서 최적화를 위해 모듈을 각각 import할 수 있도록 함

SeonHyungJo commented:
@dididy 제가 한가지 말씀을 안드렸군요. 결국 PostCSS의 autoprefixer를 사용하고 싶은거 였는데 그게 추가가 안되어있습니다. 확인부탁드릴게요

BKJang reviewed (approved):
선형님이 말씀하신 부분만 작업되면 머지되도 될 것 같습니다.
고생하셨어요 👍

voidmatcha committed:
fix: recoil 적용 부분 주석처리하여 테스트 가능하도록 함

- style import 경로 수정
- lint error 방지용 주석 추가
- Recoil 적용 부분 주석처리

voidmatcha committed:
feat: autoprefixer 설정 postcss.config.js에 추가

- 레퍼런스는 아래와 같습니다.
https://github.com/vitejs/vite/issues/1618#issuecomment-765491778
https://github.com/browserslist/browserslist

voidmatcha commented:
@SeonHyungJo 아래의 레퍼런스에 따라 autoprefixer 설정하였습니다. 다만 전역 scss import에 대한 수정이 필요해보입니다 😢

 - https://github.com/vitejs/vite/issues/1618#issuecomment-765491778
 - https://github.com/browserslist/browserslist

SeonHyungJo committed:
fix(cssnano): postCSS에는 추가되어있으나 설치되어 있지 않아 설치진행

SeonHyungJo committed:
fix(scss): 전역 scss 버그 fix

SeonHyungJo committed:
fix(postcss): autoPrefixer 설정이 예전 버전으로 되어있어 최신 문법으로 변경

browsers => overrideBrowserslist

voidmatcha commented on vite.config.ts:
```diff
@@ -19,7 +19,7 @@ export default defineConfig({
     preprocessorOptions: {
       scss: {
-        additionalData: `@import "./src/styles/index.scss"`
+        additionalData: `@import "./src/styles/index.scss";`
```
세미콜론이 이곳을 의미하는 것인지 몰랐네요.. 감사합니다!
